src/voxel.h

```diff
@@ -109,21 +109,20 @@ class VoxelArea
 		const methods
 	*/
 
-	const v3s16 &getExtent() const
+	const v3s32 &getExtent() const
 	{
 		return m_cache_extent;
 	}
 
-	/// @note `getVolume() == 0` and `getEmptyExtent()` are not identical.
 	bool hasEmptyExtent() const
 	{
-		// FIXME: shouldn't this actually be a component-wise check?
-		return m_cache_extent == v3s16(0,0,0);
+		return !m_cache_extent.X || !m_cache_extent.Y || !m_cache_extent.Z;
 	}
 
-	s32 getVolume() const
+	u32 getVolume() const
 	{
-		return (s32)m_cache_extent.X * (s32)m_cache_extent.Y * (s32)m_cache_extent.Z;
+		// FIXME: possible integer overflow here
```

Comment:
16+16+16 = 48 bits. Thus, how about using `u64` here?

Reply:
that would work but I thought "well we don't really work with areas this big so I can get away with not fixing it"...

Reply:
Right. Such area also wouldn't fit in RAM so I suppose it's OK for now.